### PR TITLE
fixes for subgraph

### PR DESCRIPTION
**Pull request by KillerByte**
* delete continuousSwap when its no longer active
* ignore 0 address events for compatibility fix

### CODE REVIEW

File: packages/subgraph/src/mapping.ts [hunk 1]
```diff
@@ -214,6 +207,19 @@ export function handleSetContinuousSwap(event: LOG_SET_FLOW): void {
   updatePoolDayData(event, 'continuous');
   updatePoolHourData(event, 'continuous');
   updateTokenDayData(tokenIn, event, 'continuous');
+
+  if(event.params.tokenRateIn.equals(BigInt.fromI32(0))) {
```
Comment (Eadword) [inline packages/subgraph/src/mapping.ts:211]: use `.equals(ZERO_BI)`

Comment (Eadword) [inline packages/subgraph/src/mapping.ts:211]: This needs to come before the time bucketed data calls.

File: packages/subgraph/src/mapping.ts [hunk 2]
```diff
@@ -214,6 +207,19 @@ export function handleSetContinuousSwap(event: LOG_SET_FLOW): void {
...
+    store.remove('ContinuousSwap', swap.id);
+    return;
+  }
+
+  swap.timestamp = event.block.timestamp;
+  swap.transaction = event.transaction.hash.toHex();
+  swap.minOut = convertTokenToDecimal(event.params.minOut, tokenOut.decimals);
+  swap.maxOut = convertTokenToDecimal(event.params.maxOut, tokenOut.decimals);
+  swap.rateIn = convertTokenToDecimal(event.params.tokenRateIn, tokenIn.decimals);
+  swap.currentRateOut = BigInt.fromI32(0).toBigDecimal();
```
Comment (Eadword) [inline packages/subgraph/src/mapping.ts:221]: Use `ZERO_BD`

Comment (Eadword) [inline packages/subgraph/src/mapping.ts:221]: You set the rate out but you neglect to update the `totalOutUntilLastSwap` amount and `timestampSinceLastSwap`.

**Review by Eadword — CHANGES_REQUESTED** [review]

File: packages/subgraph/src/mapping.ts (hunk 1 above)
Comment (KillerByte) [reply packages/subgraph/src/mapping.ts:211]: :+1:

File: packages/subgraph/src/mapping.ts [hunk 3]
```diff
@@ -197,20 +197,25 @@ export function handleSetContinuousSwap(event: LOG_SET_FLOW): void {
...
-
   tokenIn.instantSwapCount = tokenIn.instantSwapCount.plus(ONE_BI);
   tokenOut.instantSwapCount = tokenOut.instantSwapCount.plus(ONE_BI);
   pool.instantSwapCount = pool.instantSwapCount.plus(ONE_BI);
   tokenIn.save();
   tokenOut.save();
   pool.save();
 
+  if(event.params.tokenRateIn.equals(ZERO_BI)) {
+    store.remove('ContinuousSwap', swap.id);
```
Comment (Eadword) [inline packages/subgraph/src/mapping.ts:208]: Why would you delete this? It should just be marked as inactive and serve as a historical record.

**Review by Eadword — CHANGES_REQUESTED** [review]

File: packages/subgraph/src/mapping.ts (hunk 3 above)
Comment (KillerByte) [reply packages/subgraph/src/mapping.ts:208]: 2 reasons:

1. graph preserves all historical record by using the `block` property in the query. therefore it is unnecessary to keep anything historical
2. web ui wants to a list of the user's _current_ swaps. it does that by simply running a user query, and receiving the continuousSwap subdocuments. the query becomes more complicated if we start filtering by activity or other factors.

Comment (Eadword) [reply packages/subgraph/src/mapping.ts:208]: 1. then why don't we delete instant swaps as well?
2. that is why you set active to false so you can filter.

Comment (Eadword) [reply packages/subgraph/src/mapping.ts:208]: At a minimum, if you are going to change the domain this way, you need to actually change the grapql schema to reflect your decisions.

**Review by Eadword — CHANGES_REQUESTED** [review]
see other comments

**Review by Eadword — APPROVED** [review]

**Pull request merged**